dask/dataframe/core.py

```diff
@@ -5244,7 +5251,26 @@ def map_partitions(
             if not valid_divisions(divisions):
                 divisions = [None] * (dfs[0].npartitions + 1)
 
+    if has_keyword(func, "partition_info"):
+        dsk = dict(dsk)
+
+        for k, v in dsk.items():
+            vv = v
+            v = v[0]
+            [(key, task)] = v.dsk.items()  # unpack subgraph callable
```

Comment:
Are `key` and `task` ever used? It seems they're reassigned a few lines down.

Reply:
Good point, removed them.